src/src/com/microsoft/aad/adal/AuthenticationParameters.java

```diff
@@ -166,17 +167,17 @@ public static void createFromResourceUrl(Context context, final URL resourceUrl,
             public void run() {
                 HashMap<String, String> headers = new HashMap<String, String>();
                 headers.put(WebRequestHandler.HEADER_ACCEPT, WebRequestHandler.HEADER_ACCEPT_JSON);
-                HttpWebResponse webResponse = sWebRequest.sendGet(resourceUrl, headers);
 
-                if (webResponse != null) {
-                    if(webResponse.getResponseException() != null) {
```

Comment:
I like the removal for checking getResponseException, it's just wrapping the IOException, have no idea why we did it that way :disappointed: